tests/ast/data/DataFrame.to_local_iterator.test

```diff
@@ -0,0 +1,32 @@
+## TEST CASE
+
+df = session.table("test_table")
+
+rows = [row for row in df.to_local_iterator()]
+
+rows = [row for row in df.to_local_iterator(block=False)]
+
+rows = [row for row in df.to_local_iterator(statement_params={"SF_PARTNER": "FAKE_PARTNER"})]
+
+rows = [row for row in df.to_local_iterator(statement_params={"SF_PARTNER": "FAKE_PARTNER"}, case_sensitive=False)]
+
+
+## EXPECTED UNPARSER OUTPUT
+
+df = session.table("test_table")
+
+df.to_local_iterator()
```

Comment:
Do we only generate a result assignment if it's a dataframe output of the api call?

Reply:
I'm not sure what you mean by this